ItemLibrary/Smartphone.cs

```diff
@@ -20,10 +21,15 @@ public Smartphone ()
         {
         }
 
-        //will remove this method in near future
+        //find similar elements in a list
         public override List<Item> FindSimilar(List<Item> list)
+
         {
-            throw new NotImplementedException();
+            IEnumerable<Smartphone> phones = list.Cast<Smartphone>().Where(phone => phone != this && phone.RAM == this.RAM &&
+                                                                                    phone.Price >= this.Price - 100 && phone.Price <= this.Price + 100 &&
+                                                                                    phone.Storage == this.Storage &&
+                                                                                    phone.BackCameraMP.Count == this.BackCameraMP.Count);
```

Comment:
Does amount of cameras really matter? I think it should be better if the main (best) camera would be similar

Reply:
Arguably yes, since having 4 cameras makes a phone more versatile by enabling the user to take focused wide angle shots for example.

Reply:
But I could put this in a if/else so we could check if the phone has 1 or more camera, if it has 1 we can find similar phones with 1 camera and in the range of 10MP for example. What do you think?

Reply:
But e.g. OnePlus could have 3 cameras, Iphone 4 (one of them could be completely useless). So they can't be similar because of the count despite that the main camera is slightly better in OnePlus?

Reply:
10MP is a lot for phones. Should be smaller range. But Im still not sure if the count of cameras is that important. Or maybe add Camera count Range for 1, so phones with 3 or 4 cameras could be similar

Reply:
Alright I'll add the camera range 1, seems the best solution so far, we will see how this pans out in the future and change it accordingly if the need arises.